src/test/java/com/es/phoneshop/web/AddCartServletTest.java

```diff
@@ -0,0 +1,82 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.exceptions.OutOfStockException;
+import com.es.phoneshop.model.cart.Cart;
+import com.es.phoneshop.model.product.Product;
+import com.es.phoneshop.service.CartService;
+import com.es.phoneshop.service.ProductService;
+import jakarta.servlet.ServletConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.IOException;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.Locale;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.anyLong;
+
+public class AddCartServletTest {
+
+    @Mock
+    private ProductService productService;
+
+    @Mock
+    private CartService cartService;
+
+    @Mock
+    private HttpServletRequest request;
+
+    @Mock
+    private ServletConfig config;
+
+    @Mock
+    private HttpSession session;
+
+    @Mock
+    private HttpServletResponse response;
+
+    private AddCartServlet servlet;
+
+    @Before
+    public void setup() throws ServletException {
+        MockitoAnnotations.initMocks(this);
+        servlet = new AddCartServlet();
+        servlet.init(config);
+        servlet.setProductService(productService);
+        servlet.setCartService(cartService);
+        when(request.getSession()).thenReturn(session);
+    }
+
+    @Test
+    public void givenValidRequest_whenDoPost_thenAddProductToCart() throws ServletException, IOException, OutOfStockException, ParseException {
```

Comment:
when you are testing something "valid", you should cover by tests also all invalid cases